sql/sql_update.cc

```diff
@@ -529,6 +529,7 @@ int mysql_update(THD *thd,
   if (!(select && select->quick))
     status_var_increment(thd->status_var.update_scan_count);
 
+  IO_CACHE tempfile;
```

Comment:
Undefined behaviour is never safe. Probably, there is a better fix which requires a hard thinking. But this fix is no harm while in fixes and UB which is covered by `main.update` test.